src/pyskindose/rdsr_normalizer.py

```diff
@@ -217,10 +226,47 @@ def _normalize_beam_parameters(
     data_norm["kVp"] = data_parsed.KVP_kV
     data_norm["K_IRP"] = data_parsed.DoseRP_Gy * 1000
 
-    data_norm["filter_thickness_Cu"] = data_parsed.XRayFilterThicknessMaximum_mm
+    # Ad filter thincknesses...
 
-    data_norm.filter_thickness_Cu = data_norm.filter_thickness_Cu.fillna(0.0)
+    # first, replace all nans with zeros in data parsed
+    for key in [KEY_RDSR_FILTER_MIN, KEY_RDSR_FILTER_MAX]:
+        data_parsed[key] = data_parsed[key].fillna(0.0)
 
-    data_norm["filter_thickness_Al"] = [0.0] * len(data_parsed.XRayFilterThicknessMaximum_mm)
+    # Add columns for filter materials, and initialize to zero
+    for key in [
+        KEY_NORMALIZATION_FILTER_SIZE_COPPER,
+        KEY_NORMALIZATION_FILTER_SIZE_ALUMINUM,
+    ]:
+        data_norm[key] = 0.0
+
+    # for each irradiation event
+    for event_index in range(len(data_parsed)):
```

Comment:
I feel like this/the contents of this for loop could be broken out into a separate function for more clarity when reading the code.

Reply:
done